src/DynamicNLPModels.jl

```diff
@@ -732,6 +787,29 @@ function _build_J(A,B, N)
     return J
 end
 
+function _build_EF(E, F, N)
```

Comment:
how about rename these functions with indication of whether it's for sparse formulation or dense formulation? Say, `_build_dense_J2`, etc.